claims.go

```diff
@@ -344,12 +215,56 @@ func (p *Claims) validateImplID() error {
 	return nil
 }
 
-func (p *Claims) validateInstID() error {
-	if p.InstID == nil {
+func (c *Claims) instIDByProfile(profile string) *[]byte {
+	switch profile {
+	case PSA_PROFILE_1:
+		return c.LegacyInstID
+	case PSA_PROFILE_2:
+		return (*[]byte)(c.InstID)
+	default:
+		return nil
+	}
+}
+
+func (c Claims) GetInstanceID() (*[]byte, error) {
+	profile, err := c.getProfile()
+	if err != nil {
+		return nil, err
+	}
+
+	instID := c.instIDByProfile(profile)
```

Comment:
as instIDByProfile does not return any error, hence for incorrect profile set, one will return no error (nil) with nil instance ID for GetInstanceID. Perhaps should we not return an Error in GetInstanceID for incorrect profile?
This is just a suggestion..

Reply:
this is called by `Evidence::GetInstanceID` which does the check